### ISSUE TITLE
[Bug]: Crashes with fatal JRE error

### ISSUE THREAD

**Issue by ethical-haquer** [simplex-chat/simplex-chat#5099]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Platform

Linux

### OS version

Debian Testing (Trixie)

### App version

6.1.0

### Current Behavior

When attempting to run SimpleX, it crashes with a JRE fatal error. This is with the latest .DEB package on Debian Testing.

### Expected Behavior

I would expect it to run, and not crash. :)

### Steps To Reproduce

Run `/opt/simplex/bin/simplex`.
See some output as SimpleX starts.
See it (SimpleX) open, but to a black window.
Look at the terminal output and see that it crashed with a JRE fatal error.

### Relevant log output

```shell
Terminal output:
ethical_haquer@liberated-latitude:~$ /opt/simplex/bin/simplex
D: useDatabaseKey true
D: sendCmd: apiSetAppFilePaths
D: sendCmd response type cmdOk
D: sendCmd: apiSetEncryptLocalFiles
D: sendCmd response type cmdOk
D: sendCmd: showActiveUser
D: sendCmd response type chatCmdError
D: apiGetActiveUser: chatCmdError chat noActiveUser
D: sendCmd: listRemoteHosts
D: sendCmd response type remoteHostList
D: user: null
D: ChatController startReceiver
D: sendCmd: setLocalDeviceName
D: sendCmd response type cmdOk
D: startChat: started without user
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007effa5bd2681, pid=10284, tid=10380
#
# JRE version: OpenJDK Runtime Environment Corretto-17.0.12.7.1 (17.0.12+7) (build 17.0.12+7-LTS)
# Java VM: OpenJDK 64-Bit Server VM Corretto-17.0.12.7.1 (17.0.12+7-LTS, mixed mode, tiered, compressed oops, compressed class ptrs, g1 gc, linux-amd64)
# Problematic frame:
# C  [libHSzstd-0.1.3.0-203b259d81b55151fd2c9ecb432bfa701dfd8529080fb68640d12b8cbccbb2c4-ghc9.6.3.so+0xd7681]  ZSTD_freeDDict+0x11
#
# Core dump will be written. Default location: Core dumps may be processed with "/usr/lib/systemd/systemd-coredump %P %u %g %s %t %c %h" (or dumping to /home/ethical_haquer/core.10284)
#
# An error report file with more information is saved as:
# /home/ethical_haquer/hs_err_pid10284.log
#
# If you would like to submit a bug report, please visit:
#   https://github.com/corretto/corretto-17/issues/
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#
Aborted (core dumped)
```

**Comment by ethical-haquer** [comment]
Here's [hs_err_pid10284.log](https://github.com/user-attachments/files/17523466/hs_err_pid10284.log)

Let me know if you need any more info! TIA.

**Comment by ethical-haquer** [comment]
@Narasimha-sc 
Yes, the latest AppImage (6.1.1) runs fine.
I can't install the latest .DEB though, due to #5117.

**Comment by Bum-Cement** [comment]
I have the same issue with the AppImage. I'm on Fedora 40 and Fedora 41 Silverblue with OpenJDK 64-Bit Server VM Corretto-17.0.13.11.1 (17.0.13+11-LTS, mixed mode, tiered, compressed oops, compressed class ptrs, g1 gc, linux-amd64)
Problematic frame:
# C  [libc.so.6+0x76d19]  __pthread_rwlock_wrlock+0x19

**Comment by EKCKABATOP54** [comment]
I have the same issue with the AppImage compiled for aarch64. It works somehow (actually errors also appears, but at least I can see the application window) on arm server, but crashes with the same error when launched on PinePhone Pro with mobian OS

**Comment by ethical-haquer** [comment]
@EKCKABATOP54 
Did you have to compile the aarch64 AppImage? I didn't see any in the releases. TIA.

**Comment by EKCKABATOP54** [comment]
@ethical-haquer 
Yes, I compiled it from source. I had to tweak the source to get it to work. I don't have a formal guide for this, but someone wrote (not yet complete and not translated into English) build notes based on what I did and posted them in the simplex chat.

**Comment by ethical-haquer** [comment]
@EKCKABATOP54 
Thanks for clarifying. I daily drive a PPP with pmOS, but currently I use SimpleX in Waydroid. The SimpleX Linux build is able to adapt to mobile then? Like is it adaptive for smaller screens?

**Comment by ethical-haquer** [comment]
@EKCKABATOP54 
👍 
Same here, it would be really nice to use SimpleX on mobile Linux.
I'll be sure to ask about it, thanks!

As for this issue, I still can't use SimpleX on my laptop, I'm experiencing the same bug with v0.6.2.

**Comment by daic11** [comment]
> I have the same issue with the AppImage. I'm on Fedora 40 and Fedora 41 Silverblue with OpenJDK 64-Bit Server VM Corretto-17.0.13.11.1 (17.0.13+11-LTS, mixed mode, tiered, compressed oops, compressed class ptrs, g1 gc, linux-amd64) Problematic frame: # C [libc.so.6+0x76d19] __pthread_rwlock_wrlock+0x19 Both the Flatpak and the AppImage are having this problem. Both also start lagging and becoming unresponsive.
> 
> [hs_err_pid26969.log](https://github.com/user-attachments/files/17686105/hs_err_pid26969.log)
Try to reinstall java from here https://adoptium.net/temurin/releases/?version=17&arch=aarch64&os=linux and remove it by `sudo dnf remove java-17-openjdk java-17-openjdk-devel`, it fixed my problem. On aarch64 fedora dnf has java package which couldn't build simplex, so should use another one from tar website.
https://github.com/simplex-chat/simplex-chat/issues/4950#issuecomment-2563571084

**Comment by EKCKABATOP54** [comment]
Same error on latest x86_64.AppImage on arch

**Comment by somereatardedwood** [comment]
Same error on latest x86_64.AppImage on Arch

**Comment by joshbowyer** [comment]
> [@ethical-haquer](https://github.com/ethical-haquer) Yes, I also made dirty changes to the sources so that simplex would compile with the android interface. I really hope that after fixing the bug described here, I will be able to run the native build on the phone. If you are interested in reproducing the AppImage build for aarch64 with the mobile interface, you can ask about it in the official simplex chat #simplex-devs (https://simplex.chat/contact#/?v=1-4&smp=smp%3A%2F%2FPQUV2eL0t7OStZOoAsPEV2QYWt4-xilbakvGUGOItUo%3D% 40smp6.simplex.im%2FvYCRjIflKNMGYlfTkuHe4B40qSlQ0439%23%2F%3Fv%3D1-2%26dh%3DMCowBQYDK2VuAyEAHNdcqNbzXZ hyMoSBjT2R0-Eb1EPaLyUg3KZjn-kmM1w%253D%26srv%3Dbylepyau3ty4czmn77q4fglvperknl4bi2eb2fdy2bh4jxtf32kf73yd.onion&data=%7B%22type%22%3A%22group%22%2C%22groupLinkId%22%3A%22PD20tcXjw7IpkkMCfR6HLA%3D%3D%22%7D) and then I will write detailed instructions

That chat link isnt working for me, do you have those instructions in a gist or somewhere?

**Comment by Narasimha-sc** [comment]
> > [@ethical-haquer](https://github.com/ethical-haquer) Yes, I also made dirty changes to the sources so that simplex would compile with the android interface. I really hope that after fixing the bug described here, I will be able to run the native build on the phone. If you are interested in reproducing the AppImage build for aarch64 with the mobile interface, you can ask about it in the official simplex chat #simplex-devs (https://simplex.chat/contact#/?v=1-4&smp=smp%3A%2F%2FPQUV2eL0t7OStZOoAsPEV2QYWt4-xilbakvGUGOItUo%3D% 40smp6.simplex.im%2FvYCRjIflKNMGYlfTkuHe4B40qSlQ0439%23%2F%3Fv%3D1-2%26dh%3DMCowBQYDK2VuAyEAHNdcqNbzXZ hyMoSBjT2R0-Eb1EPaLyUg3KZjn-kmM1w%253D%26srv%3Dbylepyau3ty4czmn77q4fglvperknl4bi2eb2fdy2bh4jxtf32kf73yd.onion&data=%7B%22type%22%3A%22group%22%2C%22groupLinkId%22%3A%22PD20tcXjw7IpkkMCfR6HLA%3D%3D%22%7D) and then I will write detailed instructions
> 
> That chat link isnt working for me, do you have those instructions in a gist or somewhere?

you can also find simplex-devs in https://simplex.chat/directory
